dask/array/core.py

```diff
@@ -378,6 +378,10 @@ def rec_concatenate(arrays, axis=0):
         arrays = [a[None, ...] for a in arrays]
     if len(arrays) == 1:
         return arrays[0]
+    for a in arrays:
+        if hasattr(a, 'shape'):
+            if all(i == 0 for i in a.shape):
+                arrays.remove(a)
```

Comment:
I was only removing arrays with all of their shape equal to zero, your code removes arrays with any shape element equal to zero.

This was breaking some slicing tests that expected arrays with size == 0. 

I removed my changes here to `rec_concatenate`, they were only needed because I was handling ghost stuff badly.